### PR TITLE
Add support for 451 status code

### LINKED ISSUE #306: Add support for 451 status code
Announcement: https://developer.github.com/changes/2016-03-17-the-451-status-code-is-now-supported/

go-github passes along the http response object, which will already include the 451 status unchanged.  The only thing that's different is that the response now includes a new "block" field.  We could either expand the existing Error type with this field and document that it's only populated on certain types of errors, or created a new Error type entirely.  At the moment, I'm leaning toward the former, but could probably be convinced otherwise if someone has a strong opinion.

### PR DESCRIPTION
**Pull request by gmlewis**
Fixes #306.

Change-Id: I3b42c946f8255e165c3bbd83b1566675d21fb0e8

### CODE REVIEW

File: github/github_test.go [hunk 1]
```diff
@@ -538,6 +538,42 @@ func TestCheckResponse(t *testing.T) {
 	}
 }
 
+func TestCheckResponse_withBlock(t *testing.T) {
```
Comment (willnorris) [inline github/github_test.go]: maybe just add a "block" to the existing `TestCheckResponse`?  All it's really doing is verifying that the JSON is unmarshaled correctly, so splitting these into separate tests doesn't gain much.

File: github/github_test.go [hunk 2]
```diff
@@ -538,6 +538,42 @@ func TestCheckResponse(t *testing.T) {
...
+			CreatedAt: &Timestamp{time.Date(2016, 3, 17, 15, 39, 46, 0, time.UTC)},
+		},
+	}
+	if err.Block == nil && err.Block.CreatedAt == nil {
+		t.Errorf("CreatedAt = nil, want %v", want.Block.CreatedAt)
+	} else if !want.Block.CreatedAt.Equal(*err.Block.CreatedAt) {
+		t.Errorf("CreatedAt = %v, want %v", err.Block.CreatedAt, want.Block.CreatedAt)
+	}
+	// Clear times for DeepEqual to succeed.
+	err.Block.CreatedAt = nil
```
Comment (willnorris) [inline github/github_test.go]: I'm surprised this is necessary.  DeepEqual fails even for time values that are identical?

Comment (gmlewis) [inline github/github_test.go]: Thanks, Will!  (For others listening in... Will realized that `+00:00` is not the same at `time.UTC` and it should be replaced by `Z` instead.)

File: github/github_test.go (hunk 1 above)
Comment (gmlewis) [inline github/github_test.go]: Done.

File: github/github.go [hunk 3]
```diff
@@ -357,6 +357,14 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 	return response, err
 }
 
+// Block represents more detailed error information such as is returned
+// for a code 451. See https://developer.github.com/changes/2016-03-17-the-451-status-code-is-now-supported/
+// for more information.
+type Block struct {
```
Comment (willnorris) [inline github/github.go]: I worry about created a new top-level, exported type just for this very obscure kind of error.  How bout declaring this as an anonymous inner struct on the ErrorResponse.  Something like http://play.golang.org/p/tmW_cl2pGr

**Comment by willnorris** [comment]
:shipit:

**Pull request merged**